homeassistant/components/cover/__init__.py

```diff
@@ -223,6 +225,16 @@ def current_cover_tilt_position(self):
         pass
 
     @property
+    def cover_state(self):
+        """Return current state of cover."""
+        pass
+
+    @property
+    def cover_state_list(self):
+        """Return list of posible door states."""
+        pass
+
+    @property
     def state(self):
         """Return the state of the cover."""
         closed = self.is_closed
```

Comment:
This property will need to be reworked. I think the easiest way would be to just add some lines before the existing logic. These states will need to be imported at the top.
```python
if self.is_opening:
    return STATE_OPENING

if self.is_closing:
    return STATE_CLOSING
```

We'll also need to add `is_opening` and `is_closing` methods to the base class. It can be similar to [this](https://github.com/home-assistant/home-assistant/pull/8574/files#diff-aec3f5346a76b74ad2bee24c7f56f886R283) except you can just `pass` instead of raising not implemented, since they're optional.